File: sdks/typescript/services/service.ts
// file: sdks/typescript/services/service.ts
// version: 1.1.0
// guid: 16593558-ea1b-497b-9e7d-9547c7f5f04b

import { Client } from '../client/client';
import { ExampleModel } from '../models/model';

export class ExampleService {
  constructor(private readonly client: Client) {
    this.client.registerUnary('Example/Call', this.callRpc.bind(this));
  }

  private async callRpc(request: unknown): Promise<unknown> {
    const model = ExampleModel.fromJSON(request);
    model.validate();
    return { ...model.toJSON(), status: 'ok' };
  }

  async call(model: ExampleModel): Promise<unknown> {
    model.validate();
    return await this.client.callUnary('Example/Call', model.toJSON());
  }
}

File: sdks/python/services/service.py
#!/usr/bin/env python3
# file: sdks/python/services/service.py
# version: 1.0.0
# guid: c9782c55-99c7-489f-b188-8eec742f46df
"""Service wrappers for gcommon Python SDK."""
from __future__ import annotations

from typing import Any


class ExampleService:
    """Example service wrapper.

    TODO: Provide real RPC implementations.
    """

    def __init__(self) -> None:
        # TODO: receive client reference
        return None

    async def call(self, data: Any) -> Any:
        """Perform a sample RPC call."""
        # TODO: implement service call
        return None

    # TODO: Add streaming support
    # TODO: Implement error translation
    # TODO: Provide retries
    # TODO: Include metrics hooks
